panel/viewable.py

```diff
@@ -311,45 +319,26 @@ def show(self, title=None, port=0, websocket_origin=None, threaded=False,
             loop = IOLoop()
             server = StoppableThread(
                 target=self._get_server, io_loop=loop,
-                args=(port, websocket_origin, loop, open, True, title, verbose),
+                args=(port, websocket_origin, loop, open, True, title, verbose, location),
                 kwargs=kwargs)
             server.start()
         else:
             server = self._get_server(
                 port, websocket_origin, show=open, start=True,
-                title=title, verbose=verbose, **kwargs
+                title=title, verbose=verbose, location=location, **kwargs
             )
         return server
 
-    
-class Viewable(Layoutable, ServableMixin):
-    """
-    Viewable is the baseclass all objects in the panel library are
-    built on. It defines the interface for declaring any object that
-    displays itself by transforming the object(s) being wrapped into
-    models that can be served using bokeh's layout engine. The class
-    also defines various methods that allow Viewable objects to be
-    displayed in the notebook and on bokeh server.
-    """
-
-    __abstract = True
 
-    _preprocessing_hooks = []
+class Renderable(param.Parameterized):
```

Comment:
Seems like each abstract base class should have a class docstring explaining what the function of the class is.